src/TreeView/TreeView.tsx

```diff
@@ -364,32 +386,51 @@ const LinkItem: React.FC<TreeViewLinkItemProps> = ({href, onSelect, ...props}) =
   )
 }
 
-// ----------------------------------------------------------------------------
-// TreeView.LoadingItem
-
-const LoadingItem: React.FC = () => {
-  return (
-    // TODO: What aria attributes do we need to add here?
-    <Item>
-      <LeadingVisual>
-        <Spinner size="small" />
-      </LeadingVisual>
-      <Text sx={{color: 'fg.muted'}}>Loading...</Text>
-    </Item>
-  )
-}
-
-LoadingItem.displayName = 'TreeView.LoadingItem'
+LinkItem.displayName = 'TreeView.LinkItem'
 
 // ----------------------------------------------------------------------------
 // TreeView.SubTree
 
+export type SubTreeState = 'initial' | 'loading' | 'done' | 'error'
+
 export type TreeViewSubTreeProps = {
   children?: React.ReactNode
+  state?: SubTreeState
 }
 
-const SubTree: React.FC<TreeViewSubTreeProps> = ({children}) => {
-  const {isExpanded} = React.useContext(ItemContext)
+const SubTree: React.FC<TreeViewSubTreeProps> = ({state, children}) => {
+  const {announceUpdate} = React.useContext(RootContext)
+  const {itemId, isExpanded} = React.useContext(ItemContext)
+  const [isLoading, setIsLoading] = React.useState(false)
+  const {safeSetTimeout, safeClearTimeout} = useSafeTimeout()
+  const timeoutId = React.useRef<number>(0)
+
+  // Announce when content has loaded
+  React.useEffect(() => {
+    if (state === 'done') {
+      const parentItem = document.getElementById(itemId)
+
+      if (!parentItem) return
+
+      const parentName = getAccessibleName(parentItem)
+      announceUpdate(`${parentName} content loaded`)
+    }
+  }, [state, itemId, announceUpdate])
+
+  // Show loading indicator after a short delay
+  React.useEffect(() => {
+    if (state === 'loading') {
+      timeoutId.current = safeSetTimeout(() => {
```

Comment:
I think `safeSetTimeout` is a thin wrapper around `setTimeout` to make sure all timeouts get cleaned up when a component unmounts. Not sure if there are downsides to using `safeSetTimeout` here instead of `setTimeout` directly